util/src/main/java/io/kubernetes/client/util/Watch.java

```diff
@@ -125,18 +126,60 @@ private Watch(JSON json, ResponseBody body, Type watchType, Call call) {
       if (line == null) {
         throw new RuntimeException("Null response from the server.");
       }
-      try {
-        return json.deserialize(line, watchType);
-      } catch (JsonParseException ex) {
-        Type statusType = new TypeToken<Response<V1Status>>() {}.getType();
-        Response<V1Status> status = json.deserialize(line, statusType);
-        return new Response<T>(status.type, status.object);
-      }
+      return parseLine(line);
     } catch (IOException e) {
       throw new RuntimeException("IO Exception during next method.", e);
     }
   }
 
+  protected boolean isStatus(String line) throws IOException {
+    boolean found = false;
+    JsonReader reader = new JsonReader(new StringReader(line));
```

Comment:
this doubles the overhead of deserializing normal watch event 👀

Reply:
No, this is a streaming parser, so it's more like string processing. The real cost of parsing is in the object allocation and reflection. (at least that's my belief...)